OpenQA/Isotovideo/NeedleDownloader.pm

```diff
@@ -0,0 +1,105 @@
+# Copyright © 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Isotovideo::NeedleDownloader;
+
+use strict;
+use warnings;
+use Mojo::Base -base;
+use Mojo::UserAgent;
+use Mojo::URL;
+use Mojo::File;
+use File::stat;
+use Try::Tiny;
+use POSIX 'strftime';
+use testapi 'diag';
+use bmwqemu;
+
+has files_to_download => sub { [] };
+has needle_dir        => sub { needle::default_needle_dir() };
+has openqa_url        => sub { 'http://' . $bmwqemu::vars{OPENQA_URL} };
```

Comment:
As I understood the variable "OPENQA_URL" normally includes the prefix, e.g. see https://openqa.opensuse.org/tests/758251/file/vars.json . Maybe better use an URL parsing function which I am sure we have used somewhere else already.

Reply:
Ok - than the name of the variable would actually make sense. Strange that in my case the 'URL' was actually just the hostname. I'll improve that to handle both cases. We usually use `Mojo::URL` for this.